Without this patch --libs is not extracted from the pkg-config
file and thus linking fails. The proper way to fix this would be using
PKG_CHECK_MODULES from pkg.m4(7) in configure.ac.

diff -upr a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2021-09-24 00:07:38.000000000 +0200
+++ b/src/Makefile.am	2021-09-28 13:41:47.921236712 +0200
@@ -15,7 +15,7 @@ bin_PROGRAMS += %D%/openocd
 if INTERNAL_JIMTCL
 %C%_openocd_LDADD += $(top_builddir)/jimtcl/libjim.a
 else
-%C%_openocd_LDADD += -ljim
+%C%_openocd_LDADD += -ljim -lssl -lcrypto -lsqlite3 -lreadline -lz
 endif

 %C%_libopenocd_la_CPPFLAGS =
